### PR TITLE
Added @types/hot-formula-parser

### PR DESCRIPTION
**Pull request by joao-mbn**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## hot-formula-parser ([<kbd>unpkg</kbd>](https://unpkg.com/browse/hot-formula-parser@latest/))
was missing the following properties:
1. error
1. ERROR
1. ERROR_DIV_ZERO
1. ERROR_NAME
1. ERROR_NOT_AVAILABLE

<details>
<summary>as well as these 4 other properties...</summary>
<p>ERROR_NULL, ERROR_NUM, ERROR_REF, ERROR_VALUE</p>
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 6e51d107ec8360e6ce7d54226122022996f46c73
</p>

File: types/hot-formula-parser/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,555 @@
...
+    'WEIBULLDIST',
+    'WORKDAY',
+    'XIRR',
+    'XNPV',
+    'XOR',
+    'YEAR',
+    'YEARFRAC',
+];
+
+export const BUILT_IN_VARIABLES: readonly ['TRUE', 'FALSE', 'NULL'];
```
Comment (weswigham) [inline types/hot-formula-parser/index.d.ts:555]: I don't see this in the [source](https://github.com/handsontable/formula-parser/blob/develop/src/index.js).  Moreover, the docs on `setVariable` don't seem accurate - you can [easily redefine these](https://github.com/handsontable/formula-parser/blob/4ef971bfa550af631d1c79636455c4cb7730123d/src/parser.js#L81).

**Review by weswigham — CHANGES_REQUESTED** [review]
This package is also dual-mode if you [check out its package.json](https://www.npmjs.com/package/hot-formula-parser?activeTab=code), so the types need to be dual-mode to match.

File: types/hot-formula-parser/index.d.ts (hunk 1 above)
Comment (joao-mbn) [reply types/hot-formula-parser/index.d.ts]: Well noticed, thank you!
Both `setVariable` and `setFunction` return the `Parser` instance.
Built-in variables aren't exported by the package, so I removed them as well. However, I've added a mention of them in the `setVariable` method, as knowing about them might be helpful.

**Comment by joao-mbn** [comment]
> This package is also dual-mode if you [check out its package.json](https://www.npmjs.com/package/hot-formula-parser?activeTab=code), so the types need to be dual-mode to match.

I'm having trouble understanding what is required here. I've tested this locally with both import assignment `import foo = require('bar')` and `es6 imports`. Both worked as expected. I've added tests to the test file as well, and it didn't fail.

**Comment by weswigham** [comment]
Usually it means you should have an export map in the `package.json` that matches (or is very similar to) the source package, and a declaration file for each of the different formats exported in that map that is a matching format (so a .d.mts for a .mjs, or a .d.cts for a .cjs, in addition to having a matching `type: module` field in the `package.json`)

**Comment by joao-mbn** [comment]
> Usually it means you should have an export map in the `package.json` that matches (or is very similar to) the source package, and a declaration file for each of the different formats exported in that map that is a matching format (so a .d.mts for a .mjs, or a .d.cts for a .cjs, in addition to having a matching `type: module` field in the `package.json`)
> 
> TS only uses/checks that these are accurate (and match node behavior) when module+moduleResolution are node16 or higher.

Appreciate the explanation. If it's rightly implemented, I think all requested changes were addressed.

File: types/hot-formula-parser/index.d.mts [hunk 2]
```diff
@@ -0,0 +1,3 @@
+import p = require('hot-formula-parser');
+
+export = p;
```
Comment (gabritto) [inline types/hot-formula-parser/index.d.mts:3]: I think you shouldn't use `export = ...` in an `.mts`/`.d.mts` file. It's a TypeScript error to do this if `module` is set to `node16` or higher.
Please use the ESM export syntax instead.

**Review by gabritto — CHANGES_REQUESTED** [review]
Some small changes left to go.

**Review by gabritto — APPROVED** [review]

**Comment by joao-mbn** [comment]
Ready to merge

**Pull request merged**